pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java

```diff
@@ -77,4 +80,68 @@ public static boolean arrayContainsInt(int[] values, int valueToFind) {
   public static boolean arrayContainsString(String[] values, String valueToFind) {
     return ArrayUtils.contains(values, valueToFind);
   }
+
+  @ScalarFunction
+  public static String[] arraySliceString(String[] values, int start, int end) {
+    return Arrays.copyOfRange(values, start, end);
+  }
+
+  @ScalarFunction
+  public static int[] arraySliceInt(int[] values, int start, int end) {
+    return Arrays.copyOfRange(values, start, end);
+  }
+
+  @ScalarFunction
+  public static int[] arrayDistinctInt(int[] values) {
+    return Arrays.stream(values).distinct().toArray();
+  }
+
+  @ScalarFunction
+  public static String[] arrayDistinctString(String[] values) {
+    return Arrays.stream(values).distinct().toArray(String[]::new);
+  }
+
+  @ScalarFunction
+  public static int[] arrayRemoveInt(int[] values, int element) {
+    return ArrayUtils.removeElement(values, element);
+  }
+
+  @ScalarFunction
+  public static String[] arrayRemoveString(String[] values, String element) {
+    return ArrayUtils.removeElement(values, element);
+  }
+
+  @ScalarFunction
+  public static int arrayPositionInt(int[] values, int element) {
+    return ArrayUtils.indexOf(values, element);
+  }
+
+  @ScalarFunction
+  public static int arrayPositionString(String[] values, String element) {
+    return ArrayUtils.indexOf(values, element);
+  }
+
+  @ScalarFunction
+  public static int[] arrayUnionInt(int[] values1, int[] values2) {
+    Set<Integer> set = new HashSet<>(Arrays.stream(values1).boxed().collect(Collectors.toList()));
```

Comment:
Try to avoid using stream api in query execution code path because its performance is much worth than regular for loop